docs/advanced.rst

```diff
@@ -99,6 +99,71 @@ To do this, specify the ``--intermediate`` argument during bundle creation::
+and signed with a development certificate was tested sucessfully on your target
+and should become a release bundle now.
+Fo this it should get resigned with a specific release key without modifying
+the content of the bundle itself.
+
+This is what the ``resign`` command of RAUC is for::
+
+.. code-block:: sh
+
+  rauc resign --cert=<certfile> --key=<keyfile> --keyring=<keyring> <input-bundle> <output-bundle>
+
+It verifies the bundle against the given keyring, strips the old signature and
+attaches a new one based on the key and cert files provided.
+
+Switching the Keyring -- SPKI hashes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When switching form a development to a release signature, typically it is
+also required to equip the rootfs with a different keyring file.
+
+While the development system should accept both development and release
+certificates, the release system should accept release certificates only.
+
+One option to perform this exchange without having to build a new rootfs would
+be to include both a keyring for the development case as well as a keyring for
+the release case.
+
+Doing this would be possible in a slots post-install hook, for example.
+Depending on whether the bundle to install was signed with a development or a
+relase certificate, either the production or development keyring will be copied
+to the location where RAUC expects it to be.
+
+To allow comparing hashes, RAUC generates SPKI hashes (i.e. hashes over the
+entire public key information of a certificate) out of each signature
+contained in the bundles trust chain.
+The SPKI hashes are invariant over changes in signature meta data (such as the
+validity dates) while allowing to securely compare the certificate ownership.
+
+A simple call of ``rauc info`` will list the SPKI hashes for each certificate
+contained in the validated trust chain::
+
+  Certificate Chain:
+   0 Subject: /O=Test Org/CN=Test Org Release-1
+     Issuer: /O=Test Org/CN=Test Org Provisioning CA Release
+     SPKI sha256: 94:67:AB:31:08:04:3D:2D:62:D5:EE:58:D6:2F:86:7A:F2:77:94:29:9B:46:11:00:EC:D4:7B:1B:1D:42:8E:5A
+   1 Subject: /O=Test Org/CN=Test Org Provisioning CA Release
+     Issuer: /O=Test Org/CN=Test Org Provisioning CA Root
+     SPKI sha256: 47:D4:9D:73:9B:11:FB:FD:AB:79:2A:07:36:B7:EF:89:3F:34:5F:D4:9B:F3:55:0F:C1:04:E7:CC:2F:32:DB:11
+   2 Subject: /O=Test Org/CN=Test Org Provisioning CA Root
+     Issuer: /O=Test Org/CN=Test Org Provisioning CA Root
+     SPKI sha256: 00:34:F8:FE:5A:DC:3B:0D:FE:64:24:07:27:5D:14:4D:E2:39:8C:68:CC:9A:86:DD:67:03:D7:15:11:16:B4:4E
+
+A post-install hook instead can access the SPKI hashes via the environment
+variable ``RAUC_BUNDLE_SPKI_HASHES`` that will be set by RAUC when invoking the
+hook script.
+This variable will contain space-separated list of the hashes in the same order
+they are listed in ``rauc info``.
+This list can be used to define a condition in the hook for either installing
+one or the other keyring file on the target.
```

Comment:
Maybe add some small shell example here?

Reply:
Added an example post-install section for a hook script handling copying either development or release keyring to the final keyring path